src/main.js

```diff
@@ -168,9 +169,38 @@ function onPageChange() {
     });
     console.info("Monitoring", [feed]);
   }
+
+  //FB5 design
+  feed = document.querySelector("div[role=feed]:not([data-adblock-monitored])");
+  if (feed !== null) {
+    feed
+      .querySelectorAll('div[data-pagelet^="FeedUnit_"]')
+      .forEach(hideIfSponsored);
+    feedObserver = new MutationObserver((mutations) => {
+      mutations.forEach((mutation) => {
+        if (mutation.target.dataset && mutation.target.dataset.pagelet && mutation.target.dataset.pagelet.startsWith("FeedUnit_")) {
+          hideIfSponsored(mutation.target);
+        }
+        if (mutation.addedNodes) {
+          mutation.addedNodes.forEach((node) => {
+            if (node.dataset && node.dataset.pagelet && node.dataset.pagelet.startsWith("FeedUnit_")) {
+              hideIfSponsored(node);
+            }
+          });
+        }
```

Comment:
Is this needed if we loop through `mutation.target` already?

Reply:
It's needed because the mutation that updates the posts is by adding new Nodes. so we need to iterated all the new added noded and check if any of them contain the "sponsored" text

Reply:
Gotcha